dba/migrations/000001_create_tables.up.sql

```diff
@@ -1,105 +1,83 @@
 /**
- * Copyright 2022 Coinbase Global, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *  http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
+* Copyright 2022-present Coinbase Global, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
 
 CREATE EXTENSION "uuid-ossp";
-
 CREATE TYPE direction AS ENUM ('DEBIT', 'CREDIT');
 CREATE TYPE ttype AS ENUM ('ORDER', 'TRANSFER', 'CONVERT');
+CREATE TYPE TRANSACTION_STATUS as ENUM ('PENDING', 'FILLED', 'FAILED', 'CANCELED');
 
-CREATE TABLE IF NOT EXISTS account
-(
-    id           UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
-    portfolio_id UUID                           NOT NULL,
-    user_id      UUID                           NOT NULL,
-    currency     VARCHAR(64)                    NOT NULL,
-    created_at   TIMESTAMPTZ(3)   DEFAULT NOW() NOT NULL
+CREATE TABLE IF NOT EXISTS account (
+    id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+    qldb_id VARCHAR(64) NOT NULL,
+    user_id UUID NOT NULL,
+    currency VARCHAR(64) NOT NULL
 );
 
 CREATE INDEX user_accounts ON account USING HASH(user_id);
-CREATE INDEX idem_user_account ON account (portfolio_id, currency, user_id);
+CREATE INDEX idem_user_account ON account (currency, user_id);
 
-CREATE TABLE IF NOT EXISTS account_balance
-(
-    id         UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
-    account_id UUID                           NOT NULL REFERENCES account (id),
-    balance    NUMERIC          DEFAULT 0     NOT NULL CHECK (balance >= 0),
-    hold       NUMERIC          DEFAULT 0     NOT NULL CHECK (hold >= 0),
-    available  NUMERIC          DEFAULT 0     NOT NULL CHECK (available >= 0),
-    created_at TIMESTAMPTZ(3)   DEFAULT NOW() NOT NULL,
+CREATE TABLE IF NOT EXISTS account_balance (
```

Comment:
We need an updated_at field whenever the value changes

Reply:
Account Balance is still append only so I kept it as only created_at